restapi/src/main/java/io/stargate/web/resources/v2/schemas/UserDefinedTypesResource.java

```diff
@@ -0,0 +1,374 @@
+          // Permissions on a type are the same as keyspace (CreateTypeFetcher for graphQl API)
+          db.getAuthorizationService()
+              .authorizeSchemaRead(
+                  authenticatedDB.getAuthenticationSubject(),
+                  Collections.singletonList(keyspaceName),
+                  null,
+                  SourceAPI.REST,
+                  ResourceKind.TYPE);
+          // Mapping to Rest API expected output
+          UserDefinedTypeResponse udtResponse =
+              mapUdtAsReponse(authenticatedDB.getType(keyspaceName, typeName));
+          return Response.ok(
+                  Converters.writeResponse(
+                      raw
+                          ? udtResponse
+                          : new ResponseWrapper<UserDefinedTypeResponse>(udtResponse)))
+              .build();
+        });
+  }
+
+  @Timed
+  @POST
+  @ApiOperation(
+      value = "Create a user defined type (UDT)",
+      notes = "Add a user defined type (udt) in a specific keyspace.",
+      response = Map.class,
+      code = 201)
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 201, message = "Created", response = Map.class),
+        @ApiResponse(
+            code = 400,
+            message = "Bad Request, the input is not well formated",
+            response = Error.class),
+        @ApiResponse(
+            code = 401,
+            message = "Unauthorized, token is not valid or not enough permissions",
+            response = Error.class),
+        @ApiResponse(
+            code = 409,
+            message = "Conflict, the object may already exist",
+            response = Error.class),
+        @ApiResponse(code = 500, message = "Internal server error", response = Error.class)
+      })
+  public Response createType(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam(HEADER_TOKEN_AUTHENTICATION)
+          final String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(value = "", required = true) @NotNull final UserDefinedTypeAdd udtAdd,
+      @Context final HttpServletRequest request) {
+
+    return RequestHandler.handle(
+        () -> {
+          // Accessing datastore
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, getAllHeaders(request));
+          Keyspace keyspace = authenticatedDB.getDataStore().schema().keyspace(keyspaceName);
+
+          // Parameters validations, Keyspacename, type
+          if (keyspace == null) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(
+                    new Error(
+                        "keyspace does not exists", Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+          }
```

Comment:
You are checking if keyspace exists here, but doesn't do the same in `delete` method. Wouldn't be the case of using the same logic there?

Reply:
True. On every single method to be honest and externalize as a private method IMO

Reply:
There is a review needed in every resource most do not do the test.